src/hotspot/cpu/ppc/sharedRuntime_ppc.cpp

```diff
@@ -1652,9 +1652,57 @@ static void continuation_enter_cleanup(MacroAssembler* masm) {
 #endif
 
   __ ld_ptr(tmp1, ContinuationEntry::parent_cont_fastpath_offset(), R1_SP);
+  __ st_ptr(tmp1, JavaThread::cont_fastpath_offset(), R16_thread);
+
+  if (CheckJNICalls) {
+    // Check if this is a virtual thread continuation
+    Label L_skip_vthread_code;
+    __ lwz(R0, in_bytes(ContinuationEntry::flags_offset()), R1_SP);
+    __ cmpwi(CCR0, R0, 0);
+    __ beq(CCR0, L_skip_vthread_code);
+
+    Label L_no_warn;
+    __ lwz(R0, in_bytes(JavaThread::jni_monitor_count_offset()), R16_thread);
+    __ cmpwi(CCR0, R0, 0);
```

Comment:
Change to:
```
    __ ld(R0, in_bytes(JavaThread::jni_monitor_count_offset()), R16_thread);
    __ cmpdi(CCR0, R0, 0);
```
Since `_jni_monitor_count` is a double word on PPC64.